### ISSUE TITLE
Allow multi-threaded HNSW indexing for update operations

### ISSUE THREAD

**Issue by jobergum** [vespa-engine/vespa#17982]
Today put against HNSW index fields are multi-threaded (controlled by concurrency setting), while updates are single threaded. We should consider allowing updates also to be multi-threaded if possible.

**Comment by jobergum** [comment]
Similar if one changes distance-metric proton re-indexes the vector using the new distance metric but this process is also single threaded so it's faster for the user to simply re-feed from source than wait for the internal reindexing after restart. 

```
[WARNING]  [07:21:44]  Change(s) between active and new application that require restart:
                       In cluster 'docs' of type 'search':
                           Restart services of type 'searchnode' because:
                               1) Document type 'passage': Field 'mini_document_embedding' changed: change property 'distance-metric' from 'ANGULAR' to 'INNERPRODUCT'
```

**Comment by jobergum** [comment]
Added in https://github.com/vespa-engine/vespa/pull/18608

**Comment by jobergum** [comment]
Released in  7.436.10
